### DISCUSSION TITLE
Customizing Fields docs: Syntax error adding params?

### DISCUSSION THREAD

**Discussion by nimmolo** [rmosolgo/graphql-ruby#3931] [Q&A] [answered]
I'm trying to add some field authorization, copy/pasting the example in the docs:
```ruby
class Types::BaseField < GraphQL::Schema::Field
  # Pass `field ..., require_admin: true` to reject non-admin users from a given field
  def initialize(*args, **kwargs, require_admin: false, &block)
    @require_admin = require_admin
    super(*args, **kwargs, &block)
  end

  def authorized?(obj, args, ctx)
    # if `require_admin:` was given, then require the current user to be an admin
    super && (@require_admin ? ctx[:viewer]&.admin? : true)
  end
end
```
I've added `field_class(Types::BaseField)` to BaseObject, BaseInterface and BaseMutation.

However, I'm getting a syntax error when I run any query: 
```
SyntaxError: /xxx/app/graphql/types/base_field.rb:9: syntax error, unexpected tLABEL, expecting & or '&'
        ...*args, **kwargs, require_admin: false, &block)
                            ^~~~~~~~~~~~~~
```
I don't believe there are any typos; if i comment out `def initialize...` everything seems to run OK. 
The same "unexpected TLabel" error is also highlighted by VSCode's Ruby linter, verson 2.6.

**Comment by nimmolo** [comment, marked as answer]
Oh my goodness. Of course there is a syntax error... 
Added params must come immediately after `*args` and before `**kwargs`

**Reply by rmosolgo** [reply]
Sorry about that! I fixed the error in the docs in 9ddb4bd69

**Reply by nimmolo** [reply]
Haha no prob, but it did have me on my toes. :)
The param order was correct on the Visibility page with a similar block, that's how i noticed.
